### DISCUSSION TITLE
How can I create a common steps definition folder in cypress-cucumber?

### DISCUSSION THREAD

**Discussion by MatiDiez** [cypress-io/cypress#15540] [Plugins]
Hi everyone, so I have a framework built using cucumber on top of cypress. 
I'm using this version: **cypress-cucumber-preprocessor: ^4.0.1**
According to the cucumber javascript docs (confirmed by the behavior), it is mandatory for cucumber to have a folder at the same level and with the same name as the feature file, so cucumber knows where to look for the step definitions. As such, my file structure is: 

    -integration
    ---myApp
    -----users
    -------login.feature
    -------login
    ---------login.steps.js
    -------signup.feature
    -------signup
    ---------signup.steps.js
    -----someotherentity
    -------foo.feature
    -------foo
    ---------foo.steps.js

So login.feature, needs a folder at the same level called "login" where I store the step definitions and so on.

say all those feature files need these 2 steps for example

    Given('I visit the homepage', () => {cy.visit('https://myhomepage.com)})
    
    When('I click Ok', () =>{cy.get('#proceedButton').click()})

I have those 2 steps defined in all 3 files so It would make sense to have a common folder for common step definitions regarding navigation, clicks, etc. 

How can I do that here? Any ideas?

**Comment by DiegoBarjaConde** [comment]
The `cypress/integration/common` folder is often used to store step definitions that are common to various features in your Cypress with Cucumber tests.



When you use the Cucumber preprocessor with Cypress, it looks for step definitions in two places:



1. In a folder that has the same name as your `.feature` file inside the `cypress/integration` folder.

2. In the `cypress/integration/common` folder.



So, you can use the `cypress/integration/common` folder to store step definitions that you want to be available for various features. For example, you might have a step definition for "Given I am logged in" that you want to use in several features. You can put this step definition in a file in `cypress/integration/common`, and then you can use this step in any feature.



Please note that if you have step definitions with the same name in `cypress/integration/common` and in a folder that matches a `.feature` file, the step definition in the folder that matches the `.feature` file will overwrite the step definition in `cypress/integration/common`.
